Provide instructions/warning if input on homepage doesn't yield any Autocomplete results

Jkrzy opened the pull request:
For #101, ideating around possible implementations.

This approach, absent styling and content review of messaging, renders a note when:

A user has provided input in the `location` field on the homepage
Given that input, the Autocomplete API is unable to provide any location suggestions


<img width="803" alt="Screen Shot 2019-07-24 at 11 09 16 AM" src="https://user-images.githubusercontent.com/3485564/61805278-8457eb00-ae03-11e9-8caf-73ae5c76418d.png">

hursey013 commented on src/components/Input/Location.js:
```diff
@@ -5,6 +5,11 @@ import tw from 'tailwind.macro';
 import Label from './Label';
 
 class Location extends Component {
+  constructor(props) {
```
Using the binding ability of arrow functions, this constructor can be written as:
```
state = {
  noResultsWarning: false
}
```

hursey013 commented:
I think this is a definitely a great way to leverage the built-in functionality of the library. We do have similar logic occurring in the FormHomepage component itself however, which is how we are currently able to disable the submit button: https://github.com/18F/samhsa-prototype/blob/master/src/components/Form/FormHomepage.js#L46. I'm wondering what you think about leveraging that and displaying the error message there, vs having it tied into the Location input itself?

Jkrzy commented:
Thanks @hursey013! My thought was, keep the validation as close as possible to the source data and allow us to re-use this everywhere we're leveraging the `Location` component.

Jkrzy commented:
Took another shot at the styling here @hursey013 , thoughts?

hursey013 commented:
I think adding margin to the button on the homepage could work in some cases but I don't think it's 100% reliable. It may change, but right now we're using a system font stack, so the font will change depending on the OS which means a hardcoded number of pixels will not always line up correctly. This would also add margin to the top of the button on mobile when the inputs are stacked, so we'd need to do a screen size check and conditionally add the margin.

I think I'd still lean towards updating the state in the parent as being the most "reacty" and flexible option since it will allow us to have full control over the placement and styling based on whether its on the homepage or results page. Example implementation for homepage: https://github.com/18F/samhsa-prototype/commit/c36eebf49be5e16961681f014b6309ccf09b3cbb

Jkrzy commented:
Thanks @hursey013! Your approach takes the good parts here and improves quite a bit. Closing this and re-opening with your branch.

Jkrzy commented:
Updated per @hursey013 comments above. Also implemented for Results page.

I think we're ready for  content 👀.

@amandacostello @bpdesigns , can you take a glance at the messaging here?

### Desktop - homepage
<img width="705" alt="Screen Shot 2019-08-02 at 9 19 04 AM" src="https://user-images.githubusercontent.com/3485564/62372930-b43f7680-b506-11e9-81ba-b549ff6f927b.png">

hursey013 commented on src/utils/warnings.js:
```diff
@@ -0,0 +1,2 @@
+export const LOCATION_WARNING =
+  'Please enter a city or postal code and select one of the results. If you need help finding treatment, call us 1-800-662-HELP (4357)';
```
I forgot that I pull the phone number out into a constant too if you want to use it here. Example of it being used on the 404 page: https://github.com/18F/samhsa-prototype/blob/master/src/components/NoMatch.js#L22-L30

Jkrzy replied:
Great catch, thanks @hursey013 !

bpdesigns commented:
@amandacostello any thoughts on the wording above? I think it could be shorter and just read 

> Please enter a city or zip code

amandacostello commented:
> @amandacostello any thoughts on the wording above? I think it could be shorter and just read
> 
> > Please enter a city or zip code

Agree with this; since it returns results in territories, this is more inclusive.

Jkrzy commented:
Thanks @bpdesigns and @amandacostello!

One last look please @hursey013 and I think we're OK to merge.
